addon/components/pop-over/component.js

```diff
@@ -278,6 +278,22 @@ export default Component.extend({
     }
   },
 
+  getStaticParentOffset(element) {
+    let $element = jQuery(element);
```

Comment:
I'd like this to be refactored into a small independent function that looks a bit like `getScrollParent`:

```js
function getStaticParent(element) {
  let styles = getComputedStyle(element);
  if (styles.position === 'static') {
    let position = element.getBoundingClientRect();
    if (position.top > 0 || position.left > 0) {
      return element;
    }
  }
  return getStaticParent(element.parentElement);
}
```

(idk if this code actually works, or does the thing you expect it to, but I hope it provides a good direction!)

Reply:
this makes sense, but there's a bit of a wrinkle: the existing `getStaticParentOffset` function drills down from the top level through children till it gets to the component (so `html`, `body`, `some-big-container` ... `element`). 

the `getStaticParent` function you proposed moves out from `element` through parents towards `body`, `html`. 

that matters because the current `getStaticParentOffset` function will grab the biggest static element with the smallest offset. your `getStaticParent` function will grab the smallest static element with the biggest offset. so the popup will end up wayyy up and to the left.

i might be missing an obvious recursive solution, but it seems that to do it outside-in i'd need to set up a couple/few things: a `parents` attribute for an element (with a `getParents` function to get it and a probably a `clearParents` to clear it), which we can then iterate through, going from the outside in.

so two questions: am i in fact missing something obvious? and if not, does doing all that make things better or just more complex?

Reply:
That totally makes sense that my function is wrong. I think I'm trying to understand the issue that's happening with the box model here? I'm genuinely curious _why_ the pop over is being positioned incorrectly.

- Is there a reproducible case where we can test this in isolation?
- Why is the fixed element's child element being considered the CSS pane? (Is this a browser bug????)

Reply:
you're right, we should really isolate the variable that's causing this, the fix is based on an environment where there's a whole lot else at play. time for 🔬🧫⚗️🧪🔭🧬